stdlib/socket/src/main/java/org/ballerinalang/stdlib/socket/endpoint/tcp/ClientUtils.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.socket.endpoint.tcp;
+
+import org.ballerinalang.jvm.scheduling.Scheduler;
+import org.ballerinalang.jvm.values.ArrayValue;
+import org.ballerinalang.jvm.values.ErrorValue;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.jvm.values.connector.NonBlockingCallback;
+import org.ballerinalang.stdlib.socket.SocketConstants;
+import org.ballerinalang.stdlib.socket.exceptions.SelectorInitializeException;
+import org.ballerinalang.stdlib.socket.tcp.ChannelRegisterCallback;
+import org.ballerinalang.stdlib.socket.tcp.ReadPendingCallback;
+import org.ballerinalang.stdlib.socket.tcp.ReadPendingSocketMap;
+import org.ballerinalang.stdlib.socket.tcp.SelectorManager;
+import org.ballerinalang.stdlib.socket.tcp.SocketService;
+import org.ballerinalang.stdlib.socket.tcp.SocketUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AlreadyBoundException;
+import java.nio.channels.CancelledKeyException;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.NotYetConnectedException;
+import java.nio.channels.SocketChannel;
+import java.nio.channels.UnsupportedAddressTypeException;
+
+import static java.nio.channels.SelectionKey.OP_READ;
+
+/**
+ * Native function implementations of the TCP Client.
+ *
+ * @since 1.1.0
+ */
+public class ClientUtils {
```

Comment:
Shall we change the name of this class?
We used names like `ClientActions`, `ClientMethods` for the external function classes.

Please check the other places as well.

Reply:
+1. Will do.

Reply:
Fixed.